sys/linux/cgroup.txt

```diff
@@ -39,7 +42,24 @@ cgroup_paths = "./cgroup/syz0", "./cgroup.cpu/syz0", "./cgroup.net/syz0", "./cgr
+cgroup_ctrl_read = "cgroup.controllers", "cgroup.events", "cgroup.stat", "cpu.stat", "cpu.stat", "io.stat", "memory.current", "memory.events", "memory.stat", "memory.swap.current", "pids.current", "pids.events", "rdma.current", "cpuacct.stat", "cpuacct.usage_all", "cpuacct.usage_percpu", "cpuacct.usage_percpu_sys", "cpuacct.usage_percpu_user", "cpuacct.usage_sys", "cpuacct.usage_user", "cpuset.effective_cpus", "cpuset.effective_mems", "cpuset.memory_pressure", "hugetlb.2MB.limit_in_bytes", "net_prio.prioidx", "devices.list", "freezer.self_freezing", "freezer.parent_freezing"
+cgroup_ctrl_int = "cgroup.max.depth", "cgroup.max.descendants", "cpu.weight", "cpu.weight.nice", "io.bfq.weight", "io.max", "io.weight", "memory.high", "memory.low", "memory.max", "memory.swap.max", "pids.max", "rdma.max", "cgroup.clone_children", "cpuacct.usage", "cpuset.cpu_exclusive", "cpuset.cpus", "cpuset.mem_exclusive", "cpuset.mem_hardwall", "cpuset.memory_migrate", "cpuset.memory_spread_page", "cpuset.memory_spread_slab", "cpuset.mems", "cpuset.sched_load_balance", "cpuset.sched_relax_domain_level", "hugetlb.2MB.failcnt", "hugetlb.2MB.usage_in_bytes", "hugetlb.2MB.max_usage_in_bytes", "net_cls.classid", "notify_on_release"
+
+# net_prio (.ifpriomap)
+# TODO: The list of what is written to net_prio.ifpriomap can be enhanced with additional variations of network ifnames
+cgroup_netprio_ifpriomap = "lo 0", "lo 1", "lo 2", "eth0 0", "eth0 1", "eth0 2", "wlan0 0", "wlan0 1", "wlan0 2"
+openat$cgroup_netprio_ifpriomap(fd fd_cgroup, file ptr[in, string["net_prio.ifpriomap"]], flags const[O_RDWR], mode const[0]) fd_cgroup_netprio_ifpriomap
+write$cgroup_netprio_ifpriomap(fd fd_cgroup_netprio_ifpriomap, buf ptr[in, string[cgroup_netprio_ifpriomap]], len bytesize[buf])
+
+# devices (.allow, .deny)
+cgroup_devices_files = "devices.allow", "devices.deny"
+cgroup_devices_towrite = "a", "c", "b", "a *:* rwm", "a *:* r", "a *:* w", "a *:* m", "a *:* rw", "a *:* wm", "a *:* rm", "c *:* rwm", "c *:* r", "c *:* w", "c *:* m", "c *:* rw", "c *:* wm", "c *:* rm", "b *:* rwm", "b *:* r", "b *:* w", "b *:* m", "b *:* rw", "b *:* wm", "b *:* rm"
+openat$cgroup_devices(fd fd_cgroup, file ptr[in, string[cgroup_devices_files]], flags const[O_RDWR], mode const[0]) fd_cgroup_devices
+write$cgroup_devices(fd fd_cgroup_devices, buf ptr[in, string[cgroup_devices_towrite]], len bytesize[buf])
+
+# freezer (.state)
+cgroup_freezer_states = "THAWED", "FREEZING", "FROZEN"
+openat$cgroup_freezer_state(fd fd_cgroup, file ptr[in, string["freezer.state"]], flags const[O_RDWR], mode const[0]) fd_cgroup_freezer_state
+write$cgroup_freezer_state(fd fd_cgroup_freezer_state, buf ptr[in, string[cgroup_freezer_states]], len bytesize[buf])
```

Comment:
Perhaps it makes sense to add `read` syscalls for `freezer.state` as well as for `freezer.self_freezing` and `freezer.parent_freezing`. In practice fuzzing those won't probably uncover any memory corruptions, but might potentially point to some races.

Reply:
Similarly for `net_prio.prioidx` in the descriptions above.

Reply:
Besides special openat and write descriptions to describe specifically what to write to which file, the files to read are appended to [`cgroup_ctrl_read`](https://github.com/google/syzkaller/pull/1892/files#diff-0d120de677622279c29b6ec9876a2e24R45).
`freezer.state` was missing, and now added.

Reply:
Ah, got it, thanks!